specs/spec_reader_test.go

```diff
@@ -402,6 +402,54 @@ spec:
 
 	s, err := NewSpecReader([]string{f.Name()})
 	assert.NoError(t, err)
+	if t.Failed() {
+		return
+	}
+
+	assert.Equal(t, 1, len(s.Sources))
+	sp := s.Sources[0].Spec.(map[string]any)
+	assert.Equal(t, expectedCreds, sp["credentials"])
+}
+
+func TestExpandEnvNewlines(t *testing.T) {
+	expectedCreds := `-----BEGIN PRIVATE KEY-----
+MIItest
+	tabbledline
+\backslashes\here
+-----END PRIVATE KEY-----
+`
+
+	os.Setenv("TEST_ENV_CREDS3", expectedCreds)
+	cfg := []byte(`
+kind: source
+spec:
+  name: test
+  registry: local
+  path: /path/to/source
+  version: v1.0.0
+  tables: [ "some_table" ]
+  destinations: [ "test2" ]
+  spec:
+    credentials: ${TEST_ENV_CREDS3}
+    otherstuff: 2
+---
+kind: destination
+spec:
+  name: test2
+  registry: local
+  path: /path/to/dest
+`)
+
+	f, err := os.CreateTemp("", "testcase*.yaml")
+	assert.NoError(t, err)
+	defer os.Remove(f.Name())
+	assert.NoError(t, os.WriteFile(f.Name(), cfg, 0644))
+
+	s, err := NewSpecReader([]string{f.Name()})
+	assert.NoError(t, err)
+	if t.Failed() {
+		return
+	}
```

Comment:
Can we also add an assertion to check that the content is preserved properly, and not escaped after it is read?

Reply:
@hermanschaaf as in line 456?

Reply:
Ah, sorry, the GitHub diff cut it off 😄